src/main/scala/uk/gov/hmrc/testreport/plugin/ExclusionRuleReader.scala

```diff
@@ -23,34 +23,13 @@ object ExclusionRuleReader {
   type ExcludedViolations = List[AxeViolation]
   type IncludedViolations = List[AxeViolation]
 
-  /** Enhances a given path string to match anywhere within another string.
-    *
-    * The `matches` method in Scala requires the entire string to match the regex pattern. This method ensures the path
-    * can match any part of a string by:
-    *   1. Prepending `".*"` to the start if not already present, allowing for any characters (or none) before the path.
-    *      2. Appending `".*"` to the end if not already present, allowing for any characters (or none) after the path.
-    *
-    * This modification is necessary because `matches` does not perform partial matches. For simpler substring checks
-    * without regex, consider using `contains`.
-    *
-    * @param str
-    *   The path string to be enhanced for regex matching.
-    * @return
-    *   A regex pattern string that matches the given path anywhere within another string.
-    */
-  private def ensureRegexPattern(str: String): String = {
-    val withLeading  = if (!str.startsWith(".*")) s".*$str" else str
-    val withTrailing = if (!withLeading.endsWith(".*")) s"$withLeading.*" else withLeading
-    withTrailing
-  }
-
   def partitionViolations(
     rawAxeViolations: List[AxeViolation],
-    a11yExclusions: List[ExclusionRule]
+    exclusionRules: List[ExclusionRule]
   ): (ExcludedViolations, IncludedViolations) =
     rawAxeViolations
       .foldLeft((List.empty[AxeViolation], List.empty[AxeViolation])) { case ((excluded, included), violation) =>
-        a11yExclusions.find(rule => violation.url.matches(ensureRegexPattern(rule.path))) match {
+        exclusionRules.find(_.appliesTo(violation)) match {
```

Comment:
it's a small thing, but just want to say how much I like this API of `rules.appliesTo(violation)`